src/marketplace/ListingsView.js

```diff
@@ -52,33 +52,115 @@ class ListingsView extends React.Component<Props, State> {
-					jsonObjectOfSells[sale].amount,
-					jsonObjectOfSells[sale].sellerId,
-					jsonObjectOfSells[sale]._id);
-				itemsForSaleArray.push(sellingObject);
+		responsePromise.then(
+			response => response.json().then(data => {
+				if (response.status !== 200) {
+					console.log(response.status);
+					console.log(data.msg);
+					console.log(data);
+					return data;
+				}
+				else {
+					const jsonObjectOfSells = data;
+					const itemsForSaleArray = [];
+					for (const sale in jsonObjectOfSells) {
+						//Need to make sure that this sale involves a component and not a tank
+						if(getComponentType(jsonObjectOfSells[sale].itemId) != null) {
+						const typeOfItem = getComponentType(verifyComponent(jsonObjectOfSells[sale].itemId));
+							if(typeOfItem === this.props.sellerType) {
+								const sellingObject = new SaleObject(
+									jsonObjectOfSells[sale].itemId,
+									jsonObjectOfSells[sale].salePrice,
+									jsonObjectOfSells[sale].amount,
+									jsonObjectOfSells[sale].sellerId,
+									jsonObjectOfSells[sale]._id
+								);
+								itemsForSaleArray.push(sellingObject);
+							}
+						}
+					}
+					this.setState({itemsForSale:itemsForSaleArray}); 
+				}
+			})
+		).catch(
+			error => {
+				console.log('Couldnt connect to server!');
+				console.log(error);
+				return error;
 			}
-		}
-		this.setState({itemsForSale:itemsForSaleArray});  
+		); 
+	}
+
+	//This function uses the users id and gets the tanks that are active in the marketplace
+	getMarketSalesForTanks() : void  {
+		const  responsePromise: Promise<Response> = fetch('/api/marketplace/getTankMarketSales/' + this.state.userId, {
+			method: 'GET',
+			headers: {
+				'Access-Control-Allow-Origin': '*',
+				'Content-Type': 'application/json',
+				'Access-Control-Allow-Credentials': 'true',
+			},
+		});
+		responsePromise.then(
+			response => response.json().then(data => {
+				if (response.status !== 200) {
+					console.log(response.status);
+					console.log(data.msg);
+					console.log(data);
+					return data;
+				}
+				else {
+					const jsonObjectOfSells = data;
+					const itemsForSaleArray = [];
+					for (const sale in jsonObjectOfSells) {
+						//if this isn't a component it must be a tank so we can process it here
+						if(getComponentType(jsonObjectOfSells[sale].itemId._id) == null) {
+							const sellingObject = new SaleObject(
+								jsonObjectOfSells[sale].itemId.tankName,
+								jsonObjectOfSells[sale].salePrice,
+								jsonObjectOfSells[sale].amount,
+								jsonObjectOfSells[sale].sellerId,
+								jsonObjectOfSells[sale]._id);
+							itemsForSaleArray.push(sellingObject);
+						}
+					} 
+					this.setState({itemsForSale:itemsForSaleArray}); 
+				}
+			})
+		).catch(
+			error => {
+				console.log('Couldnt connect to server!');
+				console.log(error);
+				return error;
+			}
+		); 
 	}
-	
+
 	//This creates a card for every sale
 	createCards = () => {
 		const cards = []
```

Comment:
You need to change the components being listed to be in title case. It is difficult to read otherwise (shortRangeJammer vs Short Range Jammer). There is a function in armory using it.